### PR TITLE
json encoding is not lost when mocking with java api

### PR DESCRIPTION
**Pull request by twroniak**

### CODE REVIEW

**Comment by mariuszs** [comment]
:+1:  excellent :)

File: src/test/java/com/github/tomakehurst/wiremock/client/WireMockClientAcceptanceTest.java [hunk 1]
```diff
@@ -65,4 +65,17 @@ public void buildsMappingFromStaticSyntax() {
 		
 		assertThat(testClient.get("/my/new/resource").statusCode(), is(304));
 	}
+	
+	@Test
+    public void buildsMappingWithUrlOnyRequestAndResponseWithJsonBodyWithDiacriticSigns() {
```
Comment (mariuszs) [inline src/test/java/com/github/tomakehurst/wiremock/client/WireMockClientAcceptanceTest.java]: please format

File: src/main/java/com/github/tomakehurst/wiremock/client/HttpAdminClient.java [hunk 2]
```diff
@@ -165,8 +164,7 @@ private String postJsonAssertOkAndReturnBody(String url, String json, int expect
         HttpPost post = new HttpPost(url);
         try {
             if (json != null) {
-                StringEntity stringEntity = new StringEntity(json);
-                stringEntity.setContentType(APPLICATION_JSON.getMimeType());
+                StringEntity stringEntity = new StringEntity(json, APPLICATION_JSON);
```
Comment (szpak) [inline src/main/java/com/github/tomakehurst/wiremock/client/HttpAdminClient.java]: Nice catch. I didn't know that, but from the implementation point of view (in StringEntity) it is easier.

Btw, @mariuszs maybe you should report it upstream (as an possible engancement)?

Comment (mariuszs) [inline src/main/java/com/github/tomakehurst/wiremock/client/HttpAdminClient.java]: Do you mean master?

**Comment by tomakehurst** [comment]
Hey apologies for the delay getting to this.

Can you provide me with some context for this? I'm struggling to understand exactly what problem this solves.

Thanks.

**Comment by twroniak** [comment]
Hello,
Takie a look at the test I've provided, if you revert the change, the test will fail. The bug is caused by single-argument StringEntity constructor - it sets the content type to ISO-8859-1 regardless of the original encoding.

Cheers!

**Comment by tomakehurst** [comment]
Ah, OK so it's actually setting up the stub incorrectly without the encoding. Very odd.

Thanks!

**Comment by mariuszs** [comment]
@tomakehurst why orginal author is erased with your commit?

**Comment by tomakehurst** [comment]
This is standard behaviour when Github merges from a PR AFAIK. All I did was hit the Merge button on this occasion.

**Comment by tomakehurst** [comment]
Ah, that's just the extra merge commit that gets generated. Here's the original with author intact:
0305fa43cbe6aeed60997534f3fec0f35fca40cf

**Comment by mariuszs** [comment]
Ok, I see :beer:

**Pull request merged**